app/script/view_model/content/InputBarViewModel.js

```diff
@@ -75,20 +75,29 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
 
     this.hasFocus = ko.pureComputed(() => this.isEditing() || this.conversationHasFocus()).extend({notify: 'always'});
     this.hasTextInput = ko.pureComputed(() => {
-      return this.conversationEntity() ? this.conversationEntity().input().length > 0 : false;
+      return this.conversationEntity() ? this.conversationEntity().input().text.length > 0 : false;
+    });
+
+    this.inputTextOnly = ko.pureComputed({
+      read: () => this.input().text,
+      write: value => this.input({mentions: this.currentMentions, text: value}),
     });
 
     this.input = ko.pureComputed({
       read: () => {
+        let inputObject = {mentions: [], text: ''};
+
         if (this.isEditing()) {
-          return this.editInput();
+          inputObject = this.editInput();
+        } else if (this.conversationEntity() && this.conversationEntity().input()) {
+          inputObject = this.conversationEntity().input();
         }
 
-        if (this.conversationEntity()) {
-          return this.conversationEntity().input() || '';
-        }
+        this.currentMentions = inputObject.mentions.map(
```

Comment:
There is a weird relation between the `input` and the `currentMentions`. It seems to be an unecessary information duplication.

I we go for a structure that has both the text value and the mentions, I would go for fully using this structure and remove completely the `currentMentions`.
Probably I would rename the `input` observable `editedMessage`, and this will be the ground truth of our text input.
WDYT?